### DISCUSSION TITLE
nuxt 3 module federation for micro frontend

### DISCUSSION THREAD

**Discussion by PouryaNoufallah96** [nuxt/nuxt#18430] [Questions]
### Environment

dev

### Reproduction

https://github.com/PouryaNoufallah96/nuxt3-microfe

### Describe the bug

hi. can someone show me a real working sample of module federation in nuxt 3 for micro frontend development?
every sample ive seen had issues.
 i created a reproduction on git but when i run the apps i cannot get the remote component from host

**Comment by 17biubiu** [comment]
nobody discuss???

**Comment by Dodje** [comment]
it's really sad to see that the issue is not being resolved..

I've also created a reproduction repo with a quite more fresh packages and vue version.
https://github.com/Dodje/mfe-nuxt3-bug-reproduction

And with vue-to-vue the vite-plugin-federation works, with nuxt-to-vue - not :(

Actually, there is a repo with almost working example but with older version of plugin.
Why almost? Because shared -> vue is not an active option.

And if you try to connect remote which uses ui-library builded via vite, there is no way to correctly dedup vue instances.
And any component with slots crashes.
https://github.com/vuejs/core/issues/4344

And it's really not clear to whom I need to write. Vue/core? Nuxt-team? Vite? Or plugin creator..
Sadly lots of issues are being not answered/resolved for a long time.

**Comment by AloisSeckar** [comment]
What are you looking for is [Nuxt Layers](https://nuxt.com/docs/getting-started/layers). This is how to share reusable logic across more sub-projects. Sadly, I haven't played with this feature yet, so I cannot provide direct experience and guidance for using it. Just being aware of its existence.

However, there is a recent talk about Layers from **Nuxt Nation 2023**, which might help you getting some insight:
https://youtube.com/watch?v=dWXRiBQw_lE

**Reply by Dodje** [reply]
@AloisSeckar Thank for the link! I appriciate it and will try it in another scenary. It's an interesting approach, but It feels more like packages way for a few local projects/apps.

What do I need (actually my company made a request) It's to create and a shell filled with any number of separated separated "business micro-frontends" (ME).

Where the final result looks like:
First nuxt3 shell (which I totally inlove with) with 2 of 5 vue3 (maybe nuxt3 asynEtry) blocks.
And another nuxt3 shell with all of 5 blocks.

**Reply by Dodje** [reply]
Oh, I've made a little conclusion mistake because Krutie Patel provided the example with monorepo. 
As I got it there is a way to use external source of layers. Gonna try it. Thank again :)

**Reply by Dodje** [reply]
But there is still a require with module-federation approach.
